x-pack/legacy/plugins/ml/server/routes/filters.ts

```diff
@@ -0,0 +1,227 @@
+import { schema } from '@kbn/config-schema';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { wrapError } from '../client/error_wrapper';
+import { RouteInitialization } from '../new_platform/plugin';
+import { createFilterSchema, updateFilterSchema } from '../new_platform/filters_schema';
+import { FilterManager, FormFilter } from '../models/filter';
+
+// TODO - add function for returning a list of just the filter IDs.
+// TODO - add function for returning a list of filter IDs plus item count.
+function getAllFilters(context: RequestHandlerContext) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.getAllFilters();
+}
+
+function getAllFilterStats(context: RequestHandlerContext) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.getAllFilterStats();
+}
+
+function getFilter(context: RequestHandlerContext, filterId: string) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.getFilter(filterId);
+}
+
+function newFilter(context: RequestHandlerContext, filter: FormFilter) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.newFilter(filter);
+}
+
+function updateFilter(context: RequestHandlerContext, filterId: string, filter: FormFilter) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.updateFilter(filterId, filter);
+}
+
+function deleteFilter(context: RequestHandlerContext, filterId: string) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.deleteFilter(filterId);
+}
+
+export function filtersRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters Gets filters
+   * @apiName GetFilters
+   * @apiDescription Retrieves the list of filters which are used for custom rules in anomaly detection.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filters list of filters
+   */
+  router.get(
+    {
+      path: '/api/ml/filters',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getAllFilters(context);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters/:filterId Gets filter by ID
+   * @apiName GetFilterById
+   * @apiDescription Retrieves the filter with the specified ID.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter the filter with the specified ID
+   */
+  router.get(
+    {
+      path: '/api/ml/filters/{filterId}',
+      validate: {
+        params: schema.object({ filterId: schema.string() }),
```

Comment:
nit: `filterId` could be a reusable shema definition, e.g. `getModuleIdParamSchema` in `x-pack/legacy/plugins/ml/server/new_platform/modules.ts`

Reply:
Yes fair point. Will leave as-is for now as it's consistent with the approach used with the other Settings page `calendars` route.